util/validate/index.mjs

```diff
@@ -133,11 +130,11 @@ const run = ({
         .flatMap(schemas => fn(schemas) // Schema validation occurs here, then...
           .concat(openRpc(jsonSchemaSpec)
             .flatMap(orSpec => validateModules(ajvPackage(ajv, orSpec))(schemas)))))) // ...module validation
-            .collect()
-            .tap(_ => {
-              if (filesWithErrorCount > 0) {
-                // new line for easy-reading
-                console.log()
+            .filter( result => !result.valid ) // check if any results are not valid
+            .collect() // collect them into an array
+            .tap(invalidResults => {
```

Comment:
I think you can just `.map` and exit here like so:
```js
.flatMap(validateStuff)
.filter(result => !result.valid)
.collect()
.flatten()
.map(invalidThing => process.exit(-1))
```
This should still achieve the effect of running validation on everything, then exiting if any items make it through the filter. Still exits if _any_ results are invalid.

Reply:
Also, no need to get super fancy. This makes enough sense for someone to easily understand.

Reply:
cool. I also added a console.log of the error count, so the array code works better for that.